ibis/backends/dask/executor.py

```diff
@@ -0,0 +1,414 @@
+        return cls.partitionwise(mapper, kwargs, name=op.name, dtype=dtype)
+
+    @classmethod
+    def visit(cls, op: ops.ArrayColumn, cols):
+        return cls.rowwise(
+            lambda row: np.array(row, dtype=object), cols, name=op.name, dtype=object
+        )
+
+    @classmethod
+    def visit(cls, op: ops.ArrayConcat, arg):
+        dtype = PandasType.from_ibis(op.dtype)
+        return cls.rowwise(
+            lambda row: np.concatenate(row.values), arg, name=op.name, dtype=dtype
+        )
+
+    @classmethod
+    def visit(cls, op: ops.Unnest, arg):
+        arg = cls.asseries(arg)
+        mask = arg.map(lambda v: bool(len(v)), na_action="ignore")
+        return arg[mask].explode()
+
+    @classmethod
+    def visit(
+        cls, op: ops.ElementWiseVectorizedUDF, func, func_args, input_type, return_type
+    ):
+        """Execute an elementwise UDF."""
+
+        def mapper(df):
+            cols = [df[col] for col in df]
+            return func(*cols)
+
+        df, _ = cls.asframe(func_args)
+        result = df.map_partitions(mapper)
+        if op.dtype.is_struct():
+            result = result.apply(lambda row: row.to_dict(), axis=1)
+        return result
+
+    ############################# Reductions ##################################
+
+    @classmethod
+    def visit(cls, op: ops.ArgMin | ops.ArgMax, arg, key, where):
+        # TODO(kszucs): raise a warning about triggering compute()?
+        if isinstance(op, ops.ArgMin):
+            func = lambda x: x.idxmin()
+        else:
+            func = lambda x: x.idxmax()
+
+        if where is None:
+
+            def agg(df):
+                indices = func(df[key.name])
+                if isinstance(indices, (dd.Series, dd.core.Scalar)):
+                    # to support both aggregating within a group and globally
+                    indices = indices.compute()
+                return df[arg.name].loc[indices]
+        else:
+
+            def agg(df):
+                mask = df[where.name]
+                filtered = df[mask]
+                indices = func(filtered[key.name])
+                if isinstance(indices, (dd.Series, dd.core.Scalar)):
+                    # to support both aggregating within a group and globally
+                    indices = indices.compute()
+                return filtered[arg.name].loc[indices]
+
+        return agg
+
+    @classmethod
+    def visit(cls, op: ops.Correlation, left, right, where, how):
+        if where is None:
+
+            def agg(df):
+                return df[left.name].corr(df[right.name])
+        else:
+
+            def agg(df):
+                mask = df[where.name]
+                lhs = df[left.name][mask].compute()
+                rhs = df[right.name][mask].compute()
+                return lhs.corr(rhs)
+
+        return agg
+
+    @classmethod
+    def visit(cls, op: ops.Covariance, left, right, where, how):
+        # TODO(kszucs): raise a warning about triggering compute()?
```

Comment:
dask has built in covariance and correlation methods, why are they not sufficient here/why do we need to compute?

IMO anything that needs to compute like this shouldn't warn, we should just not support it unless we can do it cleanly.

Reply:
I first tried it without calling compute, but with the masking an two different dask series I've got some error I can't recall now. Deferring to a follow-up.

Reply:
https://github.com/ibis-project/ibis/issues/8187